### ISSUE TITLE
Model#reload do not reset cached relations

### ISSUE THREAD

**Issue by anykeyh** [anykeyh/clear#138]
## Problem

The cached relations are not invalidated during reloading

## How to reproduce ?

```
          u = User.create!({id: 1, first_name: "x"})
          u2 = User.create!({id: 2, first_name: "y"})

          p = Post.create! user: u, title: "Reload testing post"

          p.user.id.should eq(1)
          p.user = u2 #Change the user, DO NOT SAVE.
          p.reload # Reload the model now:
          p.user.id.should eq(1) # Cache should be invalidated
```

## Workaround

Use new object reload instead as of now:

```
  p = Post.find(1)
  # ...
  p.reload # Might have trouble


  p = Post.find(1)
  # ...
  p = Post.find(p.id) #< Force a new model, safe
```

**Comment by sam0x17** [comment]
Given that this is the issue, wouldn't it make sense to simply define `.reload` as `find(id)`?

**Comment by anykeyh** [comment]
This is already fixed but on v0.8 branch :smile:
